qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -540,6 +548,34 @@ def _pad_with_identity(self, clifford, qargs):
 
         return padded
 
+    def broadcast_arguments(self, qargs, cargs):
+        """
+        Broadcasting of the arguments.
+        This code is currently copied from Instruction.
+        This will be cleaned up when broadcasting is moved
+        to a separate model.
+
+        Args:
+            qargs (List): List of quantum bit arguments.
+            cargs (List): List of classical bit arguments.
+
+        Yields:
+            Tuple(List, List): A tuple with single arguments.
+        """
+
+        #  [[q[0], q[1]], [c[0], c[1]]] -> [q[0], c[0]], [q[1], c[1]]
+        flat_qargs = [qarg for sublist in qargs for qarg in sublist]
+        flat_cargs = [carg for sublist in cargs for carg in sublist]
+        yield flat_qargs, flat_cargs
+
+    @property
+    def definition(self):
+        """Computes and returns the circuit for this Clifford."""
+        if self._definition is None:
+            self._definition = self.to_circuit()
+
+        return self._definition
```

Comment:
> how far do we make it through a transpile call if we don't add a definition? 

I would say quite far, please see the code snippet from my answer to a previous question. 

> The long term goal is to have a transpiler pass which can synthesize Cliffords and use the transpilation context to make some intelligent choices between different synthesis methods.

I completely agree. This is also something that @ShellyGarion is interested in, as she wants to be able to choose at transpile time whether to apply a synthesis method that minimizes gate-count vs. depth and whether to apply a method that is better suited for all-to-all connectivity vs. linear-neighbor connectivity.

> can to_instruction above now return self instead of converting to a circuit and then to an instruction?

This is an interesting suggestion (that I have not tried yet). I might be wrong about this, but I am afraid that certain methods (like constructing a unitary `Operator` for a quantum circuit containing a clifford) depend on `to_operation` translating more complex objects to simpler objects.